indica/readers/abstractreader.py

```diff
@@ -84,985 +44,187 @@ def __init__(
+            "revision": revision,
         }
-        rho_poloidal_data = xr.DataArray(
-            database_results["rho_poloidal_data"], coords=coords
-        )
-        rho_poloidal_data = rho_poloidal_data.sel(t=slice(self._tstart, self._tend))
-        R_data = xr.DataArray(database_results["R_data"], coords=coords_chan)
-        z_data = xr.DataArray(database_results["z_data"], coords=coords_chan)
+        for _key, _path in quantities_paths.items():
+            _path_err = _path + "_err"
 
-        data = {}
-        for quantity in quantities:
-            if database_results.get(quantity) is None:
+            # Read quantity value
+            try:
+                q_val, q_path = self.reader_utils.get_signal(
+                    uid,
+                    instrument,
+                    _path,
+                    revision,
+                )
+            except Exception as e:
+                if self.verbose:
+                    print(f"Error reading {_path}: {e}")
+                    raise e
                 continue
-            if "_R" in quantity:
-                dims = ["t", "R"]
-            elif "_rho" in quantity:
-                dims = ["t", "rho_poloidal"]
-            elif "_data" in quantity:
-                dims = ["t", "channel"]
-            elif "R_shift" in quantity:
-                dims = ["t"]
-            else:
-                raise ValueError(f"Unknown quantity: {quantity}")
+            results[_key + "_records"] = q_path
+            results[_key] = q_val
 
-            data[quantity] = self.assign_dataarray(
-                instrument,
-                quantity,
-                database_results,
-                dims,
-                transform=None,
-            )
-
-            if "_R" in quantity:
-                data[quantity] = data[quantity].assign_coords(
-                    z=("R", database_results["z"])
+            # Read quantity error
+            try:
+                q_err, q_err_path = self.reader_utils.get_signal(
+                    uid,
+                    instrument,
+                    _path_err,
+                    revision,
                 )
+            except Exception as e:
+                q_err = np.full_like(results[_key], 0.0)
+                q_err_path = f"{e}"
+            results[_key + "_error"] = q_err
+            results[_key + "_error" + "_records"] = q_err_path
 
-            if "_data" in quantity:
-                data[quantity] = data[quantity].assign_coords(
-                    rho_poloidal=(("t", "channel"), rho_poloidal_data.data)
-                )
-                data[quantity] = data[quantity].assign_coords(
-                    z=("channel", z_data.data)
-                )
-                data[quantity] = data[quantity].assign_coords(
-                    R=("channel", R_data.data)
-                )
-        return data
+        return results
 
-    def _get_ppts(
+    def _get_thomson_scattering(
```

Comment:
I'm not really sure if we need these abstract methods

Reply:
I'd like to make sure it's understood that specific abstract methods should be implemented in machine-specific readers to refactor data structures and generate a geometry transform. I'll see what I can do to make it clear